scifin/timeseries/timeseries.py

```diff
@@ -1077,8 +1074,8 @@ def rolling_avg(self, pts=1):
         Transforms the time series into a rolling window average time series.
         """
         new_values = [self.data[x-pts+1:x].mean() for x in range(pts-1, self.nvalues, 1)]
-        new_df = pd.DataFrame(index=self.data.index[pts-1:self.nvalues], data=new_values)
-        new_ts = TimeSeries(new_df, tz=self.tz)
+        new_data = pd.Series(index=self.data.index[pts-1:self.nvalues], data=new_values)
```

Comment:
Would it be of any use to have a look into https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.rolling.html ?

Reply:
Indeed, that could bring a cleaner implementation :).

Reply:
Let's leave it for future branch improving timeseries.py.